### PR TITLE
Fix/bug

### PR DESCRIPTION
**Pull request by monik182**

### CODE REVIEW

File: javascript/react/src/hooks/downloads.ts [hunk 1]
```diff
@@ -56,7 +57,12 @@ export function useDownloadZip(
 		}
 
 		if (tables.length) {
-			const files = tables
+			const map = new Map<string, number>()
+			const renamedTables = tables.map(t => ({
+				...t,
+				name: renameDuplicatedFileName(map, t.name || t.id || 'table.csv'),
```
Comment (darthtrevino) [inline javascript/react/src/hooks/downloads.ts:63]: will we run into an issue if there are multiple files without names?

File: javascript/react/src/hooks/downloads.ts [hunk 2]
```diff
@@ -56,7 +57,12 @@ export function useDownloadZip(
...
+			}))
+			const files = renamedTables
 				.map(table => tableFile(table))
 				.filter(f => f !== null)
```
Comment (darthtrevino) [inline javascript/react/src/hooks/downloads.ts:67]: `!== null` is better expressed as `!= null` 

The reasoning here is that !== is explicitly checking for the null value, while != null will capture null and undefined.

File: javascript/utilities/src/utils/files.ts [hunk 3]
```diff
@@ -162,34 +162,45 @@ export async function getDataURL(file: BaseFile): Promise<string> {
...
-		if (count > 1) {
-			const ext = extension(name)
-			name = `${name.replace(`.${ext}`, '')} (${count}).${ext}`
-		}
-		return createBaseFile(file, { name })
-	})
+	return values.some(val => val > 1)
+}
+
+function cleanName(name: string): string {
```
Comment (darthtrevino) [inline javascript/utilities/src/utils/files.ts:199]: rename to `cleanFileName`

File: javascript/utilities/src/utils/files.ts [hunk 4]
```diff
@@ -162,34 +162,45 @@ export async function getDataURL(file: BaseFile): Promise<string> {
...
+	name: string,
+): string {
+	const next = (map.get(name) || 0) + 1
+	map.set(name, next)
+	const ext = extension(name)
+	return next > 1 ? `${name.replace(`.${ext}`, '')} (${next}).${ext}` : name
+}
+
+function hasDuplicatedNames(names: string[]): boolean {
+	const fileNames: Record<string, number> = {}
```
Comment (darthtrevino) [inline javascript/utilities/src/utils/files.ts:188]: Using a Set<string> makes more sense here. 
```
let uniqueNames = new Set<string>(names)
return uniqueNames.size< names.length
```

**Review by darthtrevino — CHANGES_REQUESTED** [review]

File: javascript/react/src/hooks/downloads.ts (hunk 1 above)
Comment (darthtrevino) [reply javascript/react/src/hooks/downloads.ts:63]: I see, the renameDupliactedFileName will resolve the name for us

Comment (monik182) [reply javascript/react/src/hooks/downloads.ts:63]: I checked, and we might. So far, when a csv file is uploaded, the id of the tableContainer object created for that file is the file name. I tried changing it to be a random string, but it broke the table select, as it showed the id of the selected file and not its name as it currently does.

**Pull request merged**